### PR TITLE
Design/single page

### PR DESCRIPTION
**Pull request by andrejak**
I think this now works more or less as intended. There are some parts that are a bit hacky, but hopefully it's good enough for the screen cast, which is blocking on the redesign.

The way selections works is that if you have a single item selected, it'll be in the URL (and vice versa). If you have none more more than 1, you're on "/" and the selection is modified in the sidebar. This required a bit of a refactor to ComponentPage and HomePage.
Currently, a broken url redirects you to the HomePage and not NotFoundPage, but I can try to change that back if desired.

Merging a 9 day old branch with current master was a pain, so please let me know if you spot something that looks like it might have gone wrong in the merge!

May-22-2019 15-17-35

### CODE REVIEW

File: packages/ui/src/components/ComponentContainer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+const selectHeight = "38px";
+const StyledTitle = styled.div`
+  font-size: ${props => props.theme.fontSizes.title};
+  font-weight: bold;
+  color: ${props => props.theme.colors.text};
+  line-height: ${selectHeight};
+`;
+
+const StyledSelect = styled(Select)`
+font-size: ${props => props.theme.fontSizes.normal}
```
Comment (coffee-cup) [inline packages/ui/src/components/ComponentContainer/index.tsx]: This should be indented

File: packages/ui/src/components/ComponentContainer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  const allStyles = styles
+    .map(x => x.style.replace(/\bbody\b/, `#${userBodyId}`))
+    .join("\n");
+
+  return (
+    <Container key={component.name}>
+      <StyledTitleContainer>
+        <StyledTitle>{component.name}</StyledTitle>
+        {themes.length > 0 && (
+          <StyledSelect
```
Comment (coffee-cup) [inline packages/ui/src/components/ComponentContainer/index.tsx]: I think the theme should be the same for all components. Not separate per component. Can be changed in another PR though.

File: packages/ui/src/components/ComponentList/index.tsx [hunk 3]
```diff
@@ -5,60 +5,167 @@ import styled from "styled-components";
...
-  font-weight: lighter;
+interface FileProps {
+  level: number;
+  selected?: boolean;
+}
+
+const StyledDirectory = styled.span<FileProps>`
+  display: block;
+  padding: ${paddings.small};
+  padding-left: calc(${paddings.medium} * (${props => props.level} + 1));
```
Comment (coffee-cup) [inline packages/ui/src/components/ComponentList/index.tsx]: I think the indent is a little large. In VSCode it is very small

image

Compare to

image

With a large file tree there might be a lot of horizontal scroll

File: packages/ui/src/components/ComponentList/index.tsx [hunk 4]
```diff
@@ -5,60 +5,167 @@ import styled from "styled-components";
...
+        level={0}
+        selected={selected}
+        select={select}
+      />
     </StyledComponentList>
   );
 };
 
-// @prodo
+const FileTree = ({
```
Comment (coffee-cup) [inline packages/ui/src/components/ComponentList/index.tsx]: Can this be moved to another file?

**Comment by andrejak** [comment]
@jake as per Samir's comment I will address your comments in the next PR, if that's okay!
